libarchive/test/test_read_disk_directory_traversals.c

```diff
@@ -1833,6 +1833,9 @@ test_parent(void)
 	}
 
 	assertChdir("..");
+	/* Change dirs mode so that test cleanup doesn't fail */
+	chmod("lock", 0755);
+	chmod("lock/lock2", 0755);
```

Comment:
I would prefer to see an `assertChmod()` function added to `test_utils` and used here.  That will detect any failures of these functions and also provide a single point where such calls can be changed if they need to become system-specific.

Reply:
@kientzle That makes sense.  I'll do that and push an update later.  Thanks for the feedback.